semgrep/semgrep/evaluation.py

```diff
@@ -172,129 +172,104 @@ def _evaluate_single_expression(
     flags: Optional[Dict[str, Any]] = None,
 ) -> Set[Range]:
 
-    assert expression.pattern_id, f"<internal error: expected pattern id: {expression}>"
```

Comment:
`assert`s are a no-op most of the time (e.g. when using `-OO` for optimization, which Python does when building wheels). The below `if` check should cover this :+1:

Reply:
I'll convert it to a `raise SemgrepError` to mimic the previous behavior :+1: